src/test/scala/mesosphere/marathon/core/task/termination/impl/KillStreamWatcherTest.scala

```diff
@@ -40,4 +44,28 @@ class KillStreamWatcherTest extends AkkaUnitTest {
 
     result.futureValue shouldBe Nil
   }
+
+  "KillStreamWatcher emits already terminated instances" in {
+
+    val empty = MarathonTestHelper.emptyInstance()
+    val unreachableInstance = empty.copy(state = empty.state.copy(condition = Condition.Unreachable))
+
+    val watcher = KillStreamWatcher.watchForKilledInstances(system.eventStream, List(unreachableInstance))
+
+    watcher.runWith(Sink.head).futureValue shouldEqual Done
+  }
+
+  "KillStreamWatcher doesn't emit anything if there are no changes" in {
+
+    val probe = TestProbe("KillStreamWatcher")
+
+    val empty = MarathonTestHelper.emptyInstance()
+    val runnningInstance = empty.copy(state = empty.state.copy(condition = Condition.Running))
+
+    val watcher = KillStreamWatcher.watchForKilledInstances(system.eventStream, List(runnningInstance))
+
+    watcher.runWith(Sink.head).onComplete(_ => probe.ref ! Done)
+    probe.expectNoMessage(100.millis)
+  }
+
```

Comment:
One more test with one _Failed_ and one _Running_ instance making sure a _Failed_ instance is emited instantly?